apiserver/facades/agent/instancemutater/instancemutater.go

```diff
@@ -384,11 +407,22 @@ func (api *InstanceMutaterAPI) machineLXDProfileInfo(m Machine) (lxdProfileInfo,
 	}, nil
 }
 
-func (api *InstanceMutaterAPI) setOneMachineCharmProfiles(machineTag string, profiles []string, canAccess common.AuthFunc) error {
+func (api *InstanceMutaterAPI) setOneMachineCharmProfiles(ctx context.Context, machineTag string, profiles []string, canAccess common.AuthFunc) error {
 	mTag, err := names.ParseMachineTag(machineTag)
 	if err != nil {
 		return errors.Trace(err)
 	}
+
+	machineUUID, err := api.machineService.GetMachineUUID(ctx, coremachine.Name(mTag.Id()))
+	if err != nil {
+		api.logger.Errorf("getting machine uuid: %w", err)
+		return errors.Trace(err)
+	}
+	if err := api.machineService.SetAppliedLXDProfileNames(ctx, machineUUID, profiles); err != nil {
+		api.logger.Errorf("setting lxd profile: %w", err)
+		return errors.Trace(err)
+	}
+	// TODO(nvinuesa): Remove this double write once we clean up instance data.
```

Comment:
question: is there a card for this TODO?

Reply:
Yes https://warthogs.atlassian.net/browse/JUJU-6703